server/file_handler.cc

```diff
@@ -34,10 +36,8 @@ RequestHandler::Status StaticFileHandler::HandleRequest(const Request& request,
 	std::string actual_uri = this->m_root_path_ + uri_no_prefix;
 
 	if (!FileIO::FileExists(actual_uri)){
-		printf("StaticfileHandler.HandleRequest: File not found:%s\n",actual_uri.c_str());
-		response->SetStatus(Response::NOT_FOUND);
-		response->AddHeader("Content-Type",http::mime_type::ContentTypeAsString(http::mime_type::CONTENT_TYPE_TEXT_HTML));
-		return FILE_NOT_FOUND;
+		printf("StaticHandler.HandleRequest: File not found:%s\n",actual_uri.c_str());
+		return NotFoundHandler_.HandleRequest(request, response);
```

Comment:
Should we handler the 404 here? I think it may be more clear to return the 404 error code to the server, and let the server call the 404 handler, instead of letting the StaticHandler do everything